src/apps/explorer/pages/AppData/DecodePage.tsx

```diff
@@ -1,13 +1,40 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import Form, { FormValidation } from '@rjsf/core'
 import { decodeAppDataSchema, FormProps, handleErrors, transformErrors } from './config'
 import DecodeAppData from 'components/AppData/DecodeAppData'
+import { TabData } from '.'
 
-const DecodePage: React.FC = () => {
-  const [formData, setFormdata] = useState<FormProps>()
-  const [isSubmitted, setIsSubmitted] = useState<boolean>(false)
-  const [disabled, setDisabled] = useState<boolean>(true)
-  const [invalidFormDataAttempted, setInvalidFormDataAttempted] = useState<boolean>(false)
+type DecodeProps = {
+  tabData: TabData
+  setTabData: React.Dispatch<React.SetStateAction<TabData>>
+}
+
+const DecodePage: React.FC<DecodeProps> = ({ tabData, setTabData }) => {
+  const { decode } = tabData
+  const [formData, setFormdata] = useState<FormProps>(decode.formData)
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(
+    decode.options.isSubmitted !== undefined ? decode.options.isSubmitted : false,
```

Comment:
Nitpick: perfect use case for [nullish coalescing operator`??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

```suggestion
    decode.options.isSubmitted ?? false,
```